### ISSUE TITLE
(1) Cryptic BNG error message if dangling bonds present; (2)  no warning by RB or BNG if rule defined without transformation.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [RuleWorld/bionetgen#47]
```
Both of these issues can be seen in the attached model file. RB identifies the 
dangling bond (rule #5) and throws an error message but when the model is run 
BNG throws the cryptic message: "Can't call method "Name" on an undefined value 
at /.../SpeciesGraph.pm line 2168."  One of two things should happen: Either 
BNG should throw a more informative error message or RB should not permit the 
model to be run.  The second alternative does not fix the problem if BNG is 
being used independently of RB.

In rule #3 a complexation is defined without bonding: P53(a,b,c) + P53(a,b,c) 
<-> P53(a,b,c).P53(a,b,c).  Therefore, this rule doesn't actually do anything.  
No error or warning message is thrown and when run BNG simply ignores it, 
listing it in the 'reaction rules' block of the NET file with no defined 
transformations.  This sort of thing should be caught.  I cannot think of a 
scenario where a modeler would intentionally define a rule that does not encode 
a transformation.
```

Original issue reported on code.google.com by `lh64@cornell.edu` on 7 Aug 2012 at 10:20

Attachments:
- [14-3-3_v2.bngl](https://storage.googleapis.com/google-code-attachments/bionetgen/issue-47/comment-0/14-3-3_v2.bngl)
